### DISCUSSION TITLE
How do I get runs by run_key in a sensor

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8425] [Q&A] [answered]
Is there a way to `get_runs` based on `run_key` in a sensor?  It doesn’t appear to be an option for `RunsFilter`.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1643916573843109?thread_ts=1643916573.843109&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
The `run_key` gets added to a tag on the run (specifically, `{"dagster/run_key": <your_run_key>} )`, so you can query for it through the tags argument on RunsFilter:

```python
from dagster import sensor
from dagster.core.storage.pipeline_run import RunsFilter


@sensor
def my_sensor(context: SensorEvaluationContext):
    runs = context.instance.get_runs(filters=RunsFilter(tags={"dagster/run_key": "foo"}))
    ...

```
